src/layer.js

```diff
@@ -145,50 +145,63 @@ export default class Layer {
    * @return {Promise} [will resolve once image src has been retrieved and dimensions calculated]
    */
   calculateSize() {
```

Comment:
sure. in fact i'm not quite happy with the way promises are nested, though do not think now it's a good moment to refactor it. will leave for better time.